### DISCUSSION TITLE
How to get a WebAsm Avalonia project deployed to IIS?

### DISCUSSION THREAD

**Discussion by TheWerewolf** [AvaloniaUI/Avalonia#11820] [Q&A]
I've searched everywhere - Avalonia Docs, forums, Google, even Microsoft's docs (BTW, congrats - Microsoft is referencing Avalonia a lot!) and I just can't make this work. To be clear, I am NOT a webdev - I'm a desktop/mobile dev - and have to figure everything out from scratch when I try something like this. I have Avalonia working just fine in macOS, Windows, iOS and Android.

I build my webasm project and it runs just fine from VS, but if I try opening the index.html file in ".\bin\Release\net7.0\browser-wasm\AppBundle" I get the intro page unrendered 
![image](https://github.com/AvaloniaUI/Avalonia/assets/12217281/c2bc5977-6750-45ef-821e-318df522693e)
with these errors in the console
![image](https://github.com/AvaloniaUI/Avalonia/assets/12217281/37301087-7fc4-49ec-a867-bfdbe00a11c9)

If I copy these assets into a folder in my IIS server and try to reference through the site's URL, same thing but with different errors in the console.
![image](https://github.com/AvaloniaUI/Avalonia/assets/12217281/53d4dfd7-5826-492d-a6e7-8c59e7461c61)

If I can get this working, I'd be happy to write a page for the docs explaining it step by step.

**Comment by maxkatz6** [comment]
> I build my webasm project and it runs just fine from VS, but if I try opening the index.html file in ".\bin\Release\net7.0\browser-wasm\AppBundle" I get the intro page unrendered

You can't just open index.html, it needs a local static files server. You can use this tool to keep it simple: https://github.com/natemcmaster/dotnet-serve

With IIS, though, I don't have much experience (and what I had is forgotten), but in general you need to search information on how to host static web sites in IIS: https://learn.microsoft.com/en-us/iis/manage/creating-websites/scenario-build-a-static-website-on-iis  https://newhelptech.wordpress.com/2018/06/20/step-by-step-how-to-host-a-static-website-using-iis-in-windows-server-2016/

**Reply by maxkatz6** [reply]
> I build my webasm project

Also, recommend using "dotnet publish" instead of "dotnet build" (or instead of simply building from the IDE).

**Reply by TheWerewolf** [reply]
I actually did it both ways  - in VS and with these steps

dotnet new avalonia.xplat -o Test.Avalonia -f net7.0
cd .\Test.Avalonia\Test.Avalonia.Browser\
dotnet workload restore
dotnet publish -c Release
dotnet serve --directory .\bin\Release\net7.0\browser-wasm\AppBundle\

The dotnet server approach on its own won't work since I want to expose it from my domain, so IIS has to manage launching the page. I have read the IIS static website docs (the same page you reference in fact), but there's a step in there I'm not seeing or getting because I don't see how to tell IIS to launch the page.

Ah well, I'll keep digging and if I can figure a way to get it to work with IIS I'll write up a step by step and you can add it to the docs if you like.

Cheers.

**Reply by maxkatz6** [reply]
> The dotnet server approach on its own won't work since I want to expose it from my domain

Yes, dotnet serve is only for local testing.
